crucible/test/helpers/Main.hs

```diff
@@ -1,20 +1,84 @@
 {-# LANGUAGE AllowAmbiguousTypes #-}
 module Main where
 
+import Control.Lens ((^.))
 import Data.List (isInfixOf)
+import Data.Maybe (fromMaybe)
 
 import Test.Hspec
 import Test.Tasty
 import Test.Tasty.Hspec (testSpec)
+import qualified Test.Tasty.HUnit as TTH
 
+import Data.Parameterized.Nonce (newIONonceGenerator)
+import Data.Parameterized.Some (Some(Some))
+import Lang.Crucible.Backend (SomeBackend(..))
+import qualified Lang.Crucible.Backend as LCB
+import Lang.Crucible.Backend.Simple (newSimpleBackend)
 import Lang.Crucible.Panic
+import Lang.Crucible.Simulator.SimError (SimErrorReason(GenericSimError))
+import What4.Expr (EmptyExprBuilderState(..))
+import What4.Expr.Builder (newExprBuilder)
+import qualified What4.Interface as W4I
+import What4.FloatMode (FloatModeRepr(FloatIEEERepr))
+import qualified What4.LabeledPred as W4L
+import qualified What4.ProgramLoc as W4P
 
 import qualified Panic as P
 
 main :: IO ()
 main =
   defaultMain =<< panicTests
 
+mkBackend :: IO (Some SomeBackend)
+mkBackend = do
+  Some nonceGen <- newIONonceGenerator
+  sym <- newExprBuilder FloatIEEERepr EmptyExprBuilderState nonceGen
+  Some . SomeBackend <$> newSimpleBackend sym
+
+assumePred :: LCB.IsSymBackend sym bak => bak -> W4I.Pred sym -> IO ()
+assumePred bak p =
+  LCB.addAssumption bak (LCB.GenericAssumption W4P.initializationLoc "" p)
```

Comment:
Worth giving a more descriptive string than "" here?

Reply:
Do you have any suggestions? The predicates being asserted in these tests are just fresh symbolic constants, they don't mean anything.

Reply:
I was just hoping to have some sort of unique, easily searchable message in the event that there is a bug and these messages are reported as part of a test assertion failure.

Reply:
Fair enough. While I remain somewhat skeptical of the value added, I'm more than willing to defer and add some strings.